Mathlib/Algebra/FreeMonoid/Basic.lean

```diff
@@ -123,6 +214,25 @@ theorem recOn_of_mul {C : FreeMonoid α → Sort*} (x : α) (xs : FreeMonoid α)
     (ih : ∀ x xs, C xs → C (of x * xs)) : @recOn α C (of x * xs) h0 ih = ih x xs (recOn xs h0 ih) :=
   rfl
 
+/-! ### Induction -/
+
+section induction_principles
+
+/-- An induction principle on free monoids, with cases for `1`, `FreeMonoid.of` and `*`. -/
+@[to_additive (attr := elab_as_elim, induction_eliminator)
+"An induction principle on free monoids, with cases for `0`, `FreeAddMonoid.of` and `+`."]
+protected theorem inductionOn {C : FreeMonoid α → Prop} (z : FreeMonoid α) (one : C 1)
+    (of : ∀ (x : α), C (FreeMonoid.of x)) (mul : ∀ (x y : FreeMonoid α), C x → C y → C (x * y)) :
+  C z := List.rec one (fun _ _ ih => mul [_] _ (of _) ih) z
+
+/-- an induction principle for free monoids more closely mirroring induction on lists -/
```

Comment:
```suggestion
/-- An induction principle for free monoids more closely mirroring induction on lists -/
```
more closely mirroring than what? Can you reference the other induction principle?

Reply:
Updated! This is just compared to FreeMonoid.inductionOn, which has cases for 1, of, and mul. Sometimes it's easier to do and induction cases like so: 1,  of a * b. This mirrors List induction, with nil and cons